tests/pytests/test_tags.py

```diff
@@ -358,3 +359,149 @@ def testEmptyTagLeak(env):
+        cmd = f'FT.SEARCH {idx}'.split(' ') + ['hello @t:{__empty}']
+        expected = [1, 'h1', ['t', '', 'text', 'hello']]
+        cmd_assert(env, cmd, expected)
+
+        # Non-empty union with another field
+        conn.execute_command('HSET', 'h2', 'text', 'love you', 't', 'Movie')
+        cmd = f'FT.SEARCH {idx}'.split(' ') + ['love | @t:{__empty}', 'SORTBY', 'text', 'ASC']
+        expected = [
+            2, 'h1', ['text', 'hello', 't', ''], 'h2', ['text', 'love you', 't', 'Movie']
+        ]
+        cmd_assert(env, cmd, expected)
+
+        # ------------------------------- APPLY --------------------------------
+        # Populate with some data that we will be able to see the `APPLYz`
+        cmd = f'FT.AGGREGATE {idx} * LOAD 1 @t APPLY upper(@t) as upper_t SORTBY 4 @t ASC @upper_t ASC'.split(' ')
+        expected = [
+            ANY, \
+            ['t', '', 'upper_t', ''], \
+            ['t', 'Movie', 'upper_t', 'MOVIE']
+        ]
+        cmd_assert(env, cmd, expected)
+
+        # ------------------------------ SORTBY --------------------------------
+        cmd = f'FT.AGGREGATE {idx} * LOAD * SORTBY 2 @t ASC'.split(' ')
+        expected = [
+            ANY, \
+            ['t', '', 'text', 'hello'], \
+            ['t', 'Movie', 'text', 'love you']
+        ]
+        cmd_assert(env, cmd, expected)
+
+        # Reverse order
+        cmd = f'FT.AGGREGATE {idx} * LOAD * SORTBY 2 @t DESC'.split(' ')
+        expected = [
+            ANY, \
+            ['t', 'Movie', 'text', 'love you'], \
+            ['t', '', 'text', 'hello']
+        ]
+        cmd_assert(env, cmd, expected)
+
+    # Create an index with a TAG field, that also indexes empty strings
+    env.expect('FT.CREATE', 'idx', 'SCHEMA', 't', 'TAG', 'EMPTY', 'text', 'TEXT').ok()
+
+    # Create a document with an empty field, corresponding to the TAG field of
+    # the index (use cluster connection - for rerouting).
+    conn = getConnectionByEnv(env)
+    conn.execute_command('HSET', 'h1', 't', '')
+
+    testHashIndex(env, conn, 'idx')
+    env.flush()
+
+    # ----------------------------- SORTABLE case ------------------------------
+    # Create an index with a SORTABLE TAG field, that also indexes empty strings
+    # TODO: The below command doesn't work, as opposed to the one below. Make it flexible so it will work.
+    # env.expect('FT.CREATE', 'idx_sortable', 'SCHEMA', 't', 'TAG', 'SORTABLE', 'EMPTY').ok()
+    env.expect('FT.CREATE', 'idx_sortable', 'SCHEMA', 't', 'TAG', 'EMPTY', 'SORTABLE', 'text', 'TEXT').ok()
+    conn.execute_command('HSET', 'h1', 't', '')
+
+    testHashIndex(env, conn, 'idx_sortable')
+    env.flush()
+
+    def testJSONIndex(env, idx):
+        # Search for a single document, via its indexed empty value
+        cmd = f'FT.SEARCH {idx}'.split(' ') + ['@t:{__empty}']
+        expected = [1, 'j1', ['$', '{"t":""}']]
+        cmd_assert(env, cmd, expected)
+
+        # More cases: TBD
```

Comment:
Need to test `EMPTY` with other `TAG` arguments such as `WITHSUFFIXTRIE`, `CASESENSITIVE`, `SEPARATOR` (even if order should be strict)

Reply:
I think the `CASESENSITIVE` case is less interesting no?
The others will be added.